utam-compiler/src/main/java/utam/compiler/grammar/UtamElement.java

```diff
@@ -383,4 +397,43 @@ private Container() {
       return new ElementContext[]{null, elementContext};
     }
   }
+
+  class Frame extends Traversal {
+    private Frame() {
+      if (filter != null
+          || isNullable != null
+          || isExternal != null
+          || elements != null
+          || shadow != null) {
+        throw new UtamError(getSupportedPropertiesErr(Type.FRAME));
+      }
+      if (!isPublic()) {
+        throw new UtamError(String.format(ERR_FRAME_SHOULD_BE_PUBLIC, name));
+      }
```

Comment:
maybe default value for isPublic should be true for frame type?
probably same would be a good idea for container BTW :) (in todo list for me)

Reply:
That's probably a good idea, but shouldn't we run that through an RFC? I mean, we can do it for both (it would be pretty easy), and I don't think anyone will object, but I'm not sure this is something we can just unilaterally decide without at least talking it over with everyone.

Reply:
sounds good, will bring it up later